integreat_cms/static/src/js/pages/collapse-subpages.ts

```diff
@@ -14,12 +16,25 @@ window.addEventListener("load", () => {
 /*
  * This function toggles all subpages of the clicked page and changes the icon
  */
-function toggleSubpages(event: Event) {
+async function toggleSubpages(event: Event) {
   event.preventDefault();
   // Get span with all data options
-  const collapseSpan = (event.target as HTMLElement).closest("span");
+  let collapseSpan = (event.target as HTMLElement).closest("span");
+  let pageId = collapseSpan.dataset.pageId;
+  let children = collapseSpan.closest("tbody").querySelectorAll(".parent-page-" + pageId);
+  if (children.length === 0) {
+    // Show loading spinner
+    collapseSpan.querySelector("svg").classList.toggle("hidden");
+    collapseSpan.querySelector(".animate-spin").classList.toggle("hidden");
+    await fetchSubpages(collapseSpan)
+    .catch(e => {
+      console.log('There has been a problem with your fetch operation: ' + e.message);
+    });
+    addDragAndDropListeners();
```

Comment:
e9dcf9c50e259166e259e231b12a29534b0a6a8d this will iterate over all ancestors and add the recently added children to their list of descendants on children expansion